src/preset.ts

```diff
@@ -1530,29 +1534,33 @@ export async function expandBuildPreset(folder: string, name: string, workspaceF
     return expandedPreset;
 }
 
-async function expandBuildPresetImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler): Promise<BuildPreset | null> {
-    let preset = getPresetByName(buildPresets(folder), name);
-    if (preset) {
-        return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, false, enableTryApplyDevEnv, errorHandler);
-    }
-
-    if (allowUserPreset) {
-        preset = getPresetByName(userBuildPresets(folder), name);
+async function expandBuildPresetImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler, inheritedByPreset?: BuildPreset): Promise<BuildPreset | null> {
+    const presetList = inheritedByPreset ? inheritedByPreset.__file!.buildPresets : buildPresets(folder);
+    const validInherit = presetList !== undefined && presetList.filter(p => p.name === name).length > 0;
+    if (validInherit) {
```

Comment:
I think for all preset types except for configure, the `validInherit` check shouldn't block the default preset, if it's a default preset then we are also valid to inherit.

Reply:
updated pr, thanks!